Bug fix : findAllLegacy senderId parameter was not converted correctly to sender_public_key database field

air1one opened the pull request:
## Proposed changes

Bug fix : findAllLegacy senderId parameter was not converted correctly to sender_public_key database field, causing "crash" (API call resulted in "internal server error"). 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/docs/contributing) documentation
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

supaiku0 commented:
@air1one  Good catch!

There's one more place where `conditions.senderPublicKey` is used:
https://github.com/air1one/core/blob/4025d0e0da2cb4c4b537cb2c6ff7b46b2fdb9882/packages/core-database-sequelize/lib/repositories/transactions.js#L34

But I'd like to suggest a slightly modified solution:
```javascript
  async findAllLegacy (params = {}) {
    if (params.senderId) {
      const senderPublicKey = this.__publicKeyfromSenderId(params.senderId)

      if (senderPublicKey) {
        params.senderPublicKey = senderPublicKey
      }
    }

    const conditions = this.__formatConditionsV1(params)
    ...
```

Following this approach makes it much harder to run into the issue in the first place.

j-a-m-l commented:
@air1one Could you review the 2 places that have this problem?
The suggestion of @supaiku0 it's a good refactor too.

air1one commented:
Yes thanks @supaiku0 @j-a-m-l will do.

air1one commented on packages/core-database-sequelize/__tests__/repositories/transactions.test.js:
```diff
@@ -99,6 +99,16 @@ describe('Transaction Repository', () => {
+      const transactionsSenderPublicKey = await repository.findAll({ senderPublicKey : senderWallet.publicKey })
+      const transactionsSenderId = await repository.findAll({ senderId : senderWallet.address })
+
+      expect(transactionsSenderId.count).toBe(transactionsSenderPublicKey.count);
```
Done.

j-a-m-l commented on packages/core-database-sequelize/__tests__/repositories/transactions.test.js:
```diff
@@ -170,6 +183,19 @@ describe('Transaction Repository', () => {
+      const transactionsSenderId = await repository.findAllLegacy({ senderId: senderWallet.address })
+
+      transactionsSenderPublicKey.rows.forEach(transactionSenderPublicKey => {
+        const transactionSenderId = transactionsSenderId.rows.find(tr => tr.id === transactionSenderPublicKey.id)
```
This wouldn't test that the number of transactions is the same, their order, etc.

air1one commented:
Thanks @j-a-m-l indeed my tests were totally wrong, sorry about that.

So after fixing it and finally having correct tests, I found a different bug which I fixed too, in __setBlockCache and __getBlockCache, it's better to understand it with the output of the failed test : 
```
Expected value to equal:
      {"block": {"height": 1, "id": "17184958558311101492"}, "blockId": "17184958558311101492", "serialized": {"data": [255, 1, 30, 2, 0, 0, 0, 0, 3, 71, 118, 189, 96, 128, 165, 4, 176, 248, 79, 141, 102, 177, 106, 242, 146, 220, 37, 58, 165, 244, 190, 139, 128, 119, 70, 168, 42, 163, 131, 189, 60, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 103, 101, 110, 101, 115, 105, 115, 95, 52, 54, 48, 69, 2, 33, 0, 192, 207, 31, 197, 71, 5, 193, 63, 112, 253, 227, 156, 85, 161, 112, 58, 76, 97, 43, 138, 145, 147, 121, 205, 91, 26, 218, 70, 76, 124, 200, 222, 2, 32, 116, 238, 98, 73, 10, 24, 64, 16, 173, 36, 24, 211, 23, 127, 242, 171, 3, 208, 45, 37, 137, 0, 1, 118, 49, 43, 144, 66, 43, 27, 214, 75], "type": "Buffer"}}
    Received:
      {"block": {"height": {"height": 1}}, "blockId": "17184958558311101492", "serialized": {"data": [255, 1, 30, 2, 0, 0, 0, 0, 3, 71, 118, 189, 96, 128, 165, 4, 176, 248, 79, 141, 102, 177, 106, 242, 146, 220, 37, 58, 165, 244, 190, 139, 128, 119, 70, 168, 42, 163, 131, 189, 60, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 103, 101, 110, 101, 115, 105, 115, 95, 52, 54, 48, 69, 2, 33, 0, 192, 207, 31, 197, 71, 5, 193, 63, 112, 253, 227, 156, 85, 161, 112, 58, 76, 97, 43, 138, 145, 147, 121, 205, 91, 26, 218, 70, 76, 124, 200, 222, 2, 32, 116, 238, 98, 73, 10, 24, 64, 16, 173, 36, 24, 211, 23, 127, 242, 171, 3, 208, 45, 37, 137, 0, 1, 118, 49, 43, 144, 66, 43, 27, 214, 75], "type": "Buffer"}}
```

Fixed it to have get/set coherent between each other. Is it OK to fix it in this PR ?

j-a-m-l commented on packages/core-database-sequelize/__tests__/repositories/transactions.test.js:
```diff
@@ -99,6 +99,21 @@ describe('Transaction Repository', () => {
+      const transactionsSenderPublicKey = await repository.findAll({ senderPublicKey: senderWallet.publicKey })
+      const transactionsSenderId = await repository.findAll({ senderId: senderWallet.address })
+
+      expect(transactionsSenderPublicKey.count).toBe(transactionsSenderId.count)
```
The last change: check the number of rows too.

We know, from the implementation, that the number of rows and count should be the same, but it should be tested too. Currently iit's evident that they're the same because the implementation is very simple, but it could be changed in the future, so testing the number of rows would be the safer.

air1one replied:
Ok I understand. Check out my last commit !
